File: src/main/java/com/shinonometn/re/ssim/models/CaptureTask.java
package com.shinonometn.re.ssim.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
public class CaptureTask {
    private String id;
    private String termCode;
    private String termName;
    private Date createDate;
    private Boolean finished = false;

    @Id
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTermCode() {
        return termCode;
    }

    public void setTermCode(String termCode) {
        this.termCode = termCode;
    }

    public String getTermName() {
        return termName;
    }

    public void setTermName(String termName) {
        this.termName = termName;
    }

    public Date getCreateDate() {
        return createDate;
    }

    public void setCreateDate(Date createDate) {
        this.createDate = createDate;
    }

    public Boolean getFinished() {
        return finished;
    }

    public void setFinished(Boolean finished) {
        this.finished = finished;
    }
}

File: src/main/java/com/shinonometn/re/ssim/models/CaptureTask.kt
package com.shinonometn.re.ssim.models

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

import java.util.Date

@Document
class CaptureTask {

    @get:Id
    var id: String? = null
    var termCode: String? = null
    var termName: String? = null
    var createDate: Date? = null
    var stage: String? = null
    var finished: Boolean? = false

    companion object {
        const val STAGE_INIT = "initialize"
        const val STAGE_IMPORT = "import"
        const val STAGE_CAPTURE = "capture"
    }
}
